CockpitPlugin/src/main/java/com/polidea/cockpitplugin/generator/DebugCockpitGenerator.kt

```diff
@@ -209,13 +275,38 @@ internal class DebugCockpitGenerator : BaseCockpitGenerator() {
         return createRemovePropertyChangeListenerMethodSpecForWrappableParam(paramName, stepValueClass, getStepListenerMapName(stepValueClass), getParametrizedCockpitStepClassName(stepValueClass))
     }
 
-    internal fun createAddPropertyChangeListenerMethodSpecForColorParam(paramName: String, color: CockpitColor) =
-            createAddPropertyChangeListenerMethodSpecForWrappableParam(paramName, mapToTypeClass(color.value), COCKPIT_COLOR_MAPPER, COLOR_LISTENER_MAP, cockpitColorClassName)
+    internal fun createDeprecatedAddPropertyChangeListenerMethodSpecForColorParam(paramName: String, color: CockpitColor) =
+            createAddPropertyChangeListenerMethodSpecForWrappableParam(
+                    paramName, mapToTypeClass(color.value),
+                    COCKPIT_COLOR_MAPPER, COLOR_LISTENER_MAP,
+                    cockpitColorClassName, deprecated = true
+            )
+
+    internal fun createAddForeverPropertyChangeListenerMethodSpecForColorParam(paramName: String, color: CockpitColor) =
+            createAddPropertyChangeListenerMethodSpecForWrappableParam(
+                    paramName, mapToTypeClass(color.value),
+                    COCKPIT_COLOR_MAPPER, COLOR_LISTENER_MAP,
+                    cockpitColorClassName
+            )
+
+    internal fun createLifecycleAcceptingAddPropertyChangeListenerMethodSpecForColorParam(paramName: String, color: CockpitColor) =
+            createAddPropertyChangeListenerMethodSpecForWrappableParam(
+                    paramName, mapToTypeClass(color.value),
+                    COCKPIT_COLOR_MAPPER, COLOR_LISTENER_MAP,
+                    cockpitColorClassName, takeLifecycleOwner = true
+            )
 
     internal fun createRemovePropertyChangeListenerMethodSpecForColorParam(paramName: String, color: CockpitColor) =
             createRemovePropertyChangeListenerMethodSpecForWrappableParam(paramName, mapToTypeClass(color.value), COLOR_LISTENER_MAP, cockpitColorClassName)
 
-    private fun createAddPropertyChangeListenerMethodSpecForWrappableParam(paramName: String, paramClass: Class<*>, mapperName: String, listenerMapName: String, wrappableClassName: TypeName): MethodSpec {
+    private fun createAddPropertyChangeListenerMethodSpecForWrappableParam(
+            paramName: String,
+            paramClass: Class<*>,
+            mapperName: String,
+            listenerMapName: String,
+            wrappableClassName: TypeName,
+            deprecated: Boolean = false,
+            takeLifecycleOwner: Boolean = false): MethodSpec { // TODO gdzie teraz jest twój bóg?
```

Comment:
What is this comment about? 😉

Reply:
@mikolak 😱

Reply:
Got stuck yesterday and left myself a note, sorry!